### DISCUSSION TITLE
How to write the equivalent of CASE in Prisma?

### DISCUSSION THREAD

**Discussion by F-LozadaL** [prisma/prisma#15985] [Q&A]
I Have this schema
`model Predio {
  id                Int         @id @default(autoincrement())
  appraise      Int      
  department String
  city              String
  owner         Propietario[]     --ManyToMany relation
  buildings     Construccion[]  --OneToMany   relation
  lands          Terreno?            --OneToOne     relation
}`

And i want to receive rows with these fields:
`id, appraise, department, city, owner(count), buildinds(count), lands(False if null True if exists)`
So far i am receiving Everything but the boolean with: 
`prisma.predio.findMany({
        include: {
            _count: {
                select: {
                    owner: true,
                    buildings: true
                }
            }
        }
    })`

I want to add another field achieving something similar to the code below in SQL.
`SELECT 
  CASE WHEN P.lands !=null 
       THEN TRUE 
       ELSE FALSE
  END AS Land
FROM Predio as P;`

Thanks.

**Comment by nurul3101** [comment]
Hello @F-LozadaL 👋 
You would need to use [Raw Query](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access) to write CASE/WHEN/THEN query.

We do have a Feature Request for adding support of CASE/WHEN/THEN: [#13529](https://github.com/prisma/prisma/issues/13529). If you could leave a 👍 to the Feature Request then it would help our product team prioritise it. Do let me know in case you face any issues while using Raw Query.

**Reply by pantharshit00** [reply]
👋 Hello @F-LozadaL,  did you perhaps have a chance to check @nurul3101's recommendation? Let us know if you still have any questions!
 
 Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
